src/Config/AuthJWT.php

```diff
@@ -0,0 +1,87 @@
+{
+    /**
+     * --------------------------------------------------------------------
+     * Name of Authenticator Header
+     * --------------------------------------------------------------------
+     * The name of Header that the Authorization token should be found.
+     * According to the specs, this should be `Authorization`, but rare
+     * circumstances might need a different header.
+     */
+    public string $authenticatorHeader = 'Authorization';
+
+    /**
+     * --------------------------------------------------------------------
+     * The Default Payload Items
+     * --------------------------------------------------------------------
+     * All JWTs will have these claims in the payload.
+     *
+     * @var array<string, string>
+     */
+    public array $defaultClaims = [
+        'iss' => '<Issuer of the JWT>',
+    ];
+
+    /**
+     * --------------------------------------------------------------------
+     * The Keys
+     * --------------------------------------------------------------------
+     * The key of the array is the key group name.
+     * The first key of the group is used for signing.
+     *
+     * @var array<string, array<int, array<string, string>>>
+     * @phpstan-var array<string, list<array<string, string>>>
+     */
+    public array $keys = [
+        'default' => [
+            // Symmetric Key
+            [
+                'kid' => '',      // Key ID. Optional if you have only one key.
+                'alg' => 'HS256', // algorithm.
+                // Set secret random string. Needs at least 256 bits for HS256 algorithm.
+                // E.g., $ php -r 'echo base64_encode(random_bytes(32));'
+                'secret' => '<Set secret random string>',
+            ],
+            // Asymmetric Key
+            // [
+            //     'kid'        => '',      // Key ID. Optional if you have only one key.
+            //     'alg'        => 'RS256', // algorithm.
+            //     'public'     => '',      // Public Key
+            //     'private'    => '',      // Private Key
+            //     'passphrase' => ''       // Passphrase
+            // ],
+        ],
+    ];
+
+    /**
+     * --------------------------------------------------------------------
+     * Time To Live (in seconds)
+     * --------------------------------------------------------------------
+     * Specifies the amount of time, in seconds, that a token is valid.
+     */
+    public int $timeToLive = HOUR;
+
+    /**
+     * --------------------------------------------------------------------
+     * Record Login Attempts
+     * --------------------------------------------------------------------
+     * Whether login attempts are recorded in the database.
+     *
+     * Valid values are:
+     * - Auth::RECORD_LOGIN_ATTEMPT_NONE
+     * - Auth::RECORD_LOGIN_ATTEMPT_FAILURE
+     * - Auth::RECORD_LOGIN_ATTEMPT_ALL
+     */
+    public int $recordLoginAttempt = Auth::RECORD_LOGIN_ATTEMPT_FAILURE;
```

Comment:
Kenjis took this time setting from me, at first I thought there was a problem with the code not registering a successful login.
Why is it not set to `Auth::RECORD_LOGIN_ATTEMPT_ALL` by default?

Reply:
The login here means the JWT is valid and authenticated. It is not the login attempt with id/password.
That is, if you set `Auth::RECORD_LOGIN_ATTEMPT_ALL`, every request to the protected pages will be recorded in the database.

See also https://github.com/codeigniter4/shield/pull/195#discussion_r886792961

Reply:
Hmm, I got it!